econml/utilities.py

```diff
@@ -697,3 +750,96 @@ def predict(self, Xt):
         t = Xt[:, -self.n_T:]
         predictions = [self.model_list[np.nonzero(t[i])[0][0]].predict(X[[i]]) for i in range(len(X))]
         return np.concatenate(predictions)
+
+
+class StatsModelsWrapper:
+    """
+    Helper class to wrap a StatsModels OLS model to conform to the sklearn API.
+
+    Parameters
+    ----------
+    fit_intercept: bool (default False)
+        Whether to fit an intercept
+
+    Attributes
+    ----------
+    fit_args: dict of str: object
+        The arguments to pass to the `OLS` regression's `fit` method.  See the
+        statsmodels documentation for more information.
+    results: RegressionResults
+        After `fit` has been called, this attribute will store the regression results.
+    """
+
+    def __init__(self, fit_intercept=False):
+        self.fit_args = {}
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit the ordinary least squares model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data
+        y : array_like, shape (n_samples, 1) or (n_samples,)
+            Target values
+        sample_weight : array_like, shape (n_samples,)
+            Individual weights for each sample
+
+        Returns
+        -------
+        self
+        """
+        assert ndim(y) == 1 or (ndim(y) == 2 and shape(y)[1] == 1)
+        y = reshape(y, (-1,))
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+        if sample_weight is not None:
+            ols = WLS(y, X, weights=sample_weight, hasconst=self.fit_intercept)
+        else:
+            ols = WLS(y, X, hasconst=self.fit_intercept)
+        self.results = ols.fit(**self.fit_args)
+        return self
+
+    def predict(self, X):
+        """
+        Predict using the model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Samples.
+
+        Returns
+        -------
+        array, shape (n_samples,)
+            Predicted values
+        """
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+        return self.results.predict(X)
+
```

Comment:
I think the statsmodels wrapper should have a coef_ and coef__interval
```python
    @property
    def coef_(self):
        return self.results.params

    def coef__interval(self, alpha):
        return self.results.conf_int(alpha=alpha)
```

Reply:
As a separate comment (most prob for next round): we might also want to surface a `_pvalue` extension enabled by inference apart from the `_interval` extension, e.g.:
```python
est.effect_pvalue(X)
est.const_marginal_effect_pvalue(X)
est.coef__pvalue()
```
which would return the p-value of the hypothesis test of whether that quantity is non-zero (i.e. is the effect at X non-zero, is the const_marginal_effect at X non-zero, is the coefficient non-zero). This is not directly parseable from the confidence intervals and is a very useful inference quantity. This should be very easy to implement for statsmodels and for bootstrap.

Reply:
In fact this pvalue interface might be even better if it could include a test value, as opposed to always testing for zero, i.e. pvalue(X, value=v), which tests the hypothesis that the effect at X is equal to v. Similar maybe to the t_test interface of the GLM of statsmodels:
http://www.statsmodels.org/devel/generated/statsmodels.genmod.generalized_linear_model.PredictionResults.html?highlight=predictionresults#statsmodels.genmod.generalized_linear_model.PredictionResults

Reply:
Latest revision contains `coef_` and `coef__interval`.

Let's consider p-value extensions separately to make sure that our design is fully baked before we implement it.